improver/spotdata/spot_extraction.py

```diff
@@ -171,13 +191,49 @@ def check_for_unique_id(neighbour_cube: Cube) -> Optional[Tuple[ndarray, str]]:
         else:
             return (unique_id_coord.points, unique_id_coord.name())
 
+    def get_aux_coords(
+        self, coordinate_cube: Cube, diagnostic_cube: Cube
+    ) -> Tuple[List[AuxCoord]]:
+        """
+        Extract scalar and non-scalar auxilliary coordinates from the diagnostic
+        cube. Multi-dimensional auxilliary coordinates have the relevant points
+        and bounds extracted for each site and a 1-dimensional coordinate is
+        constructed that can be associated with the spot-index coordinate.
+
+        Args:
+            coordinate_cube:
+                A cube containing the x and y grid coordinates for the grid
+                point neighbours to spot sites.
+            diagnostic_cube:
+                The cube from which auxilliary coordinates will be taken.
+
+        Returns:
+            - list of scalar coordinates
+            - list of non-scalar, multi-dimensional coordinates reshaped into
+              1-dimensional coordinates.
+        """
+        scalar_coords = []
+        nonscalar_coords = []
+        for coord in diagnostic_cube.aux_coords:
+            if coord.ndim > 1:
+                coord_points, coord_bounds = self.extract_data(
+                    coordinate_cube, diagnostic_cube, coordinate=coord.name()
+                )
+                nonscalar_coords.append(
+                    coord.copy(points=coord_points, bounds=coord_bounds)
+                )
+            elif coord.points.size == 1:
```

Comment:
Should this just be `else`?

Reply:
I've not added any handling for a case where there is a non-scalar aux coordinate which is not 2-dimensional, e.g.

```
air_temperature / (K)               (percentile: 5; projection_y_coordinate: 970; projection_x_coordinate: 1042)
     Dimension coordinates:
          percentile                           x                           -                             -
          projection_y_coordinate              -                           x                             -
          projection_x_coordinate              -                           -                             x
     Auxiliary coordinates:
          kittens                              x                           -                             -
     Scalar coordinates:
          forecast_period: 0 seconds
          forecast_reference_time: 2019-01-01 00:00:00
          height: 1.5 m
          time: 2019-01-01 00:00:00

```
In this case the coordinate `kittens` is associated with the percentile coordinate. This percentile coordinate will continue to exist in the spot-output, so the `kittens` coordinate could also be copied across and remain associated with it (arguably it should be). However, we have no use case for this at the moment as we don't have any diagnostics of this form, and it will require an additional list and noting of coordinate associations to make it work. This seemed like extra code for a completeness of functionality that we don't need, so I haven't done it. Hence, here we must check that it is a true scalar coordinate by checking the size of the coordinate.